pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -271,38 +295,32 @@ func (r *ReconcileMachineHealthCheck) getMachinesFromMHC(mhc mhcv1beta1.MachineH
 }
 
 func (r *ReconcileMachineHealthCheck) getMachineFromNode(node corev1.Node) (*mapiv1.Machine, error) {
-	machineKey, ok := node.Annotations[machineAnnotationKey]
-	if !ok {
-		glog.V(4).Infof("no machine annotation for node %s", node.GetName())
-		return nil, nil
-	}
-	glog.V(4).Infof("Node %s is annotated with machine %s", node.GetName(), machineKey)
-
-	namespace, machineName, err := cache.SplitMetaNamespaceKey(machineKey)
-	if err != nil {
-		return nil, fmt.Errorf("machine name has wrong format %v", machineKey)
-	}
-	machine := &mapiv1.Machine{}
-	if err = r.client.Get(
+	machineList := &mapiv1.MachineList{}
+	if err := r.client.List(
 		context.TODO(),
-		types.NamespacedName{
-			Namespace: namespace,
-			Name:      machineName,
-		},
-		machine,
+		machineList,
+		client.MatchingFields{machineNodeNameIndex: node.Name},
```

Comment:
I guess this doesn't matter that much, but most of these have a function that returns the type, so I think you're kind of expected to instead write: `client.MatchingField(machineNodeNameIndex, node.Name)`

Reply:
yeh I started that way but `MatchingField()` is now deprecated and it recommends to user as above

Reply:
Ah, I missed that. I thought it was the other way around! Cool